modules/core/src/main/scala/org/scalasteward/core/nurture/NurtureAlg.scala

```diff
@@ -113,6 +116,40 @@ final class NurtureAlg[F[_]](implicit
       }
     } yield result
 
+  def closeObsoletePullRequests(data: UpdateData): F[Unit] = {
+    val crossDependency = data.update match {
+      case Update.Single(crossDependency, _, _) =>
+        crossDependency
+      case Update.Group(crossDependencies, _) =>
+        crossDependencies.head
+    }
+    for {
+      _ <- logger.info(s"Looking to close obsolete PRs for ${data.update.name}")
+      prsToClose <- pullRequestRepository.getObsoleteOpenPullRequests(
+        data.repo,
+        crossDependency,
+        data.update.nextVersion
+      )
+      dataPerId = prsToClose.flatMap { case (url, data) =>
+        vcsExtraAlg
+          .extractPRIdFromUrls(config.vcsType, url)
+          .tupleRight(data)
+          .toList
+      }
+      _ <- dataPerId.traverse { case (id, pullRequestData) =>
+        closePR(data.repo, pullRequestData, id).handleErrorWith { case NonFatal(ex) =>
+          logger.warn(ex)(s"Failed to close obsolete PR #$id for ${data.updateBranch.name}")
+        }
+      }
+
+    } yield ()
+  }
+
+  private def closePR(repo: Repo, prUpdate: Update, id: Int): F[Unit] =
+    vcsApiAlg.closePullRequest(repo, id) *> logger.info(
+      s"Closed a PR for ${prUpdate.show}"
+    )
```

Comment:
I think we should also set the `PullRequestState` to `Closed` in `PullRequestRepository` for the closed PR.

Reply:
Ahh that's true I forgot to update the store, will do